src/utils/use-stats-data.ts

```diff
@@ -24,29 +24,89 @@ export function useStatsData() {
 
   useEffect(() => {
     const fetchData = async () => {
-      const [holders, volume, iq, lp, social, ep] = await Promise.all([
-        getTokenHolders(),
-        getVolume(),
-        getIQ(),
-        getLPs(),
-        getSocialData(),
-        getEpData(),
-      ])
```

Comment:
@SidharthK2 rather i think its better to add timeout to these fetch calls ? ie we can halt execution of a promise inside promise.all if it goes above say 4s

Reply:
Interesting idea, let's go with that.